protocol/parser/parser.go

```diff
@@ -122,105 +123,120 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
 	return parsed, parsedSuccessfully
 }
 
-func ParseBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
-	parsedBlockInfo, parsedSuccessfully := parseInputFromParamsWithGenericParsers(rpcInput, genericParsers)
-	if parsedSuccessfully {
-		return parsedBlockInfo
-	}
-	if parsedBlockInfo == nil {
-		parsedBlockInfo = NewParsedInput()
-	}
-
-	parsedBlockInfo.parsedBlock = func() int64 {
-		// first we try to parse the value with the block parser
-		result, err := parse(rpcInput, blockParser, PARSE_PARAMS)
-		if err != nil || result == nil {
-			utils.LavaFormatDebug("ParseBlockFromParams - parse failed",
-				utils.LogAttr("error", err),
-				utils.LogAttr("result", result),
-				utils.LogAttr("blockParser", blockParser),
-				utils.LogAttr("rpcInput", rpcInput),
+func ParseRawBlock(rpcInput RPCInput, parsedInput *ParsedInput, defaultValue string) {
```

Comment:
Basically just a wrapper for `rpcInput.ParseBlock(rawBlock)`.
The result is then stored in `ParsedInput`.
On a failed parse, it will parse the default value and store it in `ParsedInput`.

Reply:
Added comment here a00402bde70748f08989ee266f91557a843d4e2d